src/main/java/gregtech/integration/exnihilo/recipes/ExNihiloRecipes.java

```diff
@@ -0,0 +1,234 @@
+import gregtech.integration.exnihilo.ExNihiloModule;
+import gregtech.loaders.recipe.MetaTileEntityLoader;
+
+import net.minecraft.init.Blocks;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidUtil;
+
+import exnihilocreatio.ModBlocks;
+import exnihilocreatio.ModItems;
+import exnihilocreatio.compatibility.jei.crucible.CrucibleRecipe;
+import exnihilocreatio.compatibility.jei.sieve.SieveRecipe;
+import exnihilocreatio.registries.manager.ExNihiloRegistryManager;
+import exnihilocreatio.registries.types.Siftable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static gregtech.api.recipes.RecipeMaps.*;
+import static gregtech.api.unification.material.Materials.*;
+import static gregtech.api.unification.ore.OrePrefix.stick;
+import static gregtech.api.unification.ore.OrePrefix.stone;
+import static gregtech.common.blocks.BlockSteamCasing.SteamCasingType.BRONZE_HULL;
+import static gregtech.integration.exnihilo.ExNihiloModule.*;
+import static gregtech.loaders.recipe.CraftingComponent.*;
+
+public class ExNihiloRecipes {
+
+    public static void registerGTRecipes() {
+        // Machine Recipes
+        MetaTileEntityLoader.registerMachineRecipe(SIEVES, "CPC", "FMF", "OSO", 'M', HULL, 'C', CIRCUIT, 'O', CABLE,
+                'F', CONVEYOR, 'S', new ItemStack(ModBlocks.sieve), 'P', PISTON);
+        ModHandler.addShapedRecipe(true, "steam_sieve_bronze", STEAM_SIEVE_BRONZE.getStackForm(), "BPB", "BMB", "BSB",
+                'B', new UnificationEntry(OrePrefix.pipeSmallFluid, Materials.Bronze), 'M',
+                MetaBlocks.STEAM_CASING.getItemVariant(BRONZE_HULL), 'S', new ItemStack(ModBlocks.sieve), 'P',
+                Blocks.PISTON);
+        ModHandler.addShapedRecipe(true, "steam_sieve_steel", STEAM_SIEVE_STEEL.getStackForm(), "BPB", "WMW", "BBB",
+                'B', new UnificationEntry(OrePrefix.pipeSmallFluid, Materials.TinAlloy), 'M',
+                STEAM_SIEVE_BRONZE.getStackForm(), 'W', new UnificationEntry(OrePrefix.plate, Materials.WroughtIron),
+                'P', new UnificationEntry(OrePrefix.plate, Materials.Steel));
+    }
+
+    public static void registerCraftingRecipes() {
+        // Pebbles
+        ModHandler.addShapedRecipe("basalt", OreDictUnifier.get(stone, Basalt, 1), "PP", "PP", 'P',
+                new ItemStack(ExNihiloModule.GTPebbles, 1, 0));
+        ModHandler.addShapedRecipe("black_granite", OreDictUnifier.get(stone, GraniteBlack, 1), "PP", "PP", 'P',
+                new ItemStack(ExNihiloModule.GTPebbles, 1, 1));
+        ModHandler.addShapedRecipe("marble", OreDictUnifier.get(stone, Marble, 1), "PP", "PP", 'P',
+                new ItemStack(ExNihiloModule.GTPebbles, 1, 2));
+        ModHandler.addShapedRecipe("red_granite", OreDictUnifier.get(stone, GraniteRed, 1), "PP", "PP", 'P',
+                new ItemStack(ExNihiloModule.GTPebbles, 1, 3));
```

Comment:
Instead of going into the smelted forms, the pebbles should compact into the cobble stone forms when possible, to match the stone pebbles compacting into cobblestone. So the black granite pebbles should compact into black granite cobblestone, and same with the other pebbles. Both for the crafting recipe and the compacting recipe.